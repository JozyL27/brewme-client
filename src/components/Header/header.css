.brewHeader {
    color: black;
    transition-duration: 0.4s;
}

.brewHeader:hover {
    color: rgb(135,206,250);
}

nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: row;
}

h1 {
    font-size: 20px;
    text-align: center;
    text-decoration: none;
    transition-duration: 0.4s;
    padding: 10px;
}

h1:hover {
    color: rgb(135,206,250);
    text-shadow: 1px 1px 2px black, 0 0 25px rgb(135,206,250), 0 0 5px rgb(135,206,250);
}

.inLink {
    font-weight: bold;
    color: white;
    background-color: black;
    transition-duration: 0.4s;
    border: 1px solid black;
    padding: 8px;
    border-radius: 8px;
    font-size: 10px;
}

.inLink:hover {
    color: white;
    background-color: red;
}

.outLink {
    font-weight: bold;
    background-color: rgb(135,206,250);
    color: black;
    transition-duration: 0.4s;
    border: 1px solid black;
    padding: 8px;
    border-radius: 8px;
    font-size: 10px;
}

.outLink:hover {
    color: black;
    background-color: #77dd77;
}

.loggedIn, .renderLogin {
    padding: 10px;
}

/* tablets */
@media screen and (min-width: 768px) {
    .brewHeader {
        font-size: 30px;
    }

    .outLink, .inLink {
        font-size: 14px;
    }
}

/* laptops */
@media screen and (min-width: 1024px) {
    .brewHeader {
        font-size: 30px;
    }

    .loggedIn, .renderLogin {
        font-size: 14px;
    }
}