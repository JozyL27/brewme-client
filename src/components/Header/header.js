import React from 'react'
import TokenService from '../../services/token-service'
import { Link } from 'react-router-dom'
import './header.css'

export default class Header extends React.Component {
    handleLogoutClick = () => {
        TokenService.clearAuthToken()
    }

    renderLogoutLink() {
        return (
            <div className='loggedIn'>
                <Link
                onClick={this.handleLogoutClick}
                to='/'>
                Logout
                </Link>
            </div>
        )
    }

    renderLoginLink() {
        return (
            <div className='renderLogin'>
                <Link
                to='/login'>
                Login
                </Link>
            </div>
        )
    }

    render() {
    return (
        <>
        <nav>
            <h1>
                <Link to='/'>
                BrewMe
                </Link>
            </h1>
            {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </nav>
        </>
    )
    }
}