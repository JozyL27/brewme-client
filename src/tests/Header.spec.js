import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Header from '../components/Header/header'


describe(`header component`, () => {

    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <BrowserRouter>
            <Header />
            </BrowserRouter>,
            div
        )
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`renders UI as expected`, () => {
        const tree = renderer
            .create(<BrowserRouter>
            <Header />
            </BrowserRouter>)
            .toJSON()
            expect(tree).toMatchSnapshot()
    })
    
})