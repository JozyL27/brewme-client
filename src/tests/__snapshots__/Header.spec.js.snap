// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component renders UI as expected 1`] = `
<nav>
  <h1>
    <a
      className="brewHeader"
      href="/"
      onClick={[Function]}
    >
      BrewMe
    </a>
  </h1>
  <div
    className="renderLogin"
  >
    <a
      className="outLink"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</nav>
`;
